fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneQueryMultiFieldSearchClause.java

```diff
@@ -64,7 +70,22 @@ public Query bind(@Nonnull FDBRecordStoreBase<?> store, @Nonnull Index index, @N
         final LuceneAnalyzerCombinationProvider analyzerSelector = LuceneAnalyzerRegistryImpl.instance().getLuceneAnalyzerCombinationProvider(index, LuceneAnalyzerType.FULL_TEXT);
         final String[] fieldNames = LuceneScanParameters.indexTextFields(index, store.getRecordMetaData()).toArray(new String[0]);
         final String searchString = isParameter ? (String)context.getBinding(search) : search;
-        final QueryParser parser = new LuceneOptimizedMultiFieldQueryParser(fieldNames, analyzerSelector.provideQueryAnalyzer(searchString).getAnalyzer());
+        final LuceneOptimizedMultiFieldQueryParser parser = new LuceneOptimizedMultiFieldQueryParser(fieldNames, analyzerSelector.provideQueryAnalyzer(searchString).getAnalyzer());
+        Map<String, PointsConfig> pointsConfigMap = new HashMap<>();
+        final Collection<RecordType> recordTypes = store.getRecordMetaData().recordTypesForIndex(index);
+        for (RecordType type : recordTypes) {
+            LuceneIndexExpressions.getDocumentFieldDerivations(index.getRootExpression(), type.getDescriptor()).forEach((key, value) -> {
+                if (pointsConfigMap.containsKey(key)) {
+                    //TODO(bfines) this may not be correct
+                    throw new RecordCoreException("Duplicate key in Index");
```

Comment:
`PointsConfig` seems to have a reasonable `equals` method, so maybe allow if the same so that multiple record types will work (although they are presently disallowed elsewhere).